### ISSUE TITLE
[Question] Distrobox mounts /nix into containers

### ISSUE THREAD

**Issue by A-Lovett** [89luca89/distrobox#837]
I was surprised to find that when I make a distrobox container while running NixOS, my /nix directory gets mounted into the container. I can imagine that leads to various advantages, but it also causes disadvantages. Notably, when I try to set up nix's home manager within the container, it screws up my home manager setup outside the container.

Could anyone tell me why distrobox mounts /nix, and perhaps give me an idea of how to avoid mounting it, so I can set up an entirely unrelated nix/home-manager install within the container?

I appreciate it.

**Comment by shymega** [comment]
I can confirm I have encountered this too. It's very confusing, because my Nix programs and container programs all mix together. I think the scripts just iterate over `/proc/mounts`, and mount them into the container...... maybe? Would be nice to see a CLI or env var to disable certain mountpoints.

**Comment by A-Lovett** [comment]
> I can confirm I have encountered this too. It's very confusing, because my Nix programs and container programs all mix together. I think the scripts just iterate over `/proc/mounts`, and mount them into the container...... maybe? Would be nice to see a CLI or env var to disable certain mountpoints.

Yeah, things seem generally wonky. I'm finding that distrobox generally works until I call a distrobox command from inside a container--which I don't do on purpose, but I do it accidentally fairly often. After that, the system gets into a weird state where I get an error message every time I call any podman command, even outside the container ("ERRO[0000] User-selected graph driver "overlay" overwritten by graph driver "vfs" from database - delete libpod local files to resolve "). The only way to fix it is to nuke ~/.local/share/containers and restart. And then I accidentally call distrobox inside a container again.

For now, it seems like the options are either to use distrobox or to use NixOS, but not both. I'm leaning towards NixOS at the moment because it's new and exciting (for me), and because there also seem to be issues with accessing Cuda inside distrobox/podman containers: https://github.com/89luca89/distrobox/issues/804

**Comment by shymega** [comment]
Looking at `mount` _inside_ the container, it's got all my mounts from the host, inside the container. I've had to give up using Distrobox, and instead, I'm using Vagrant/Docker directly.

**Comment by A-Lovett** [comment]
> Looking at `mount` _inside_ the container, it's got all my mounts from the host, inside the container. I've had to give up using Distrobox, and instead, I'm using Vagrant/Docker directly.

I also gave up and implemented the functionality I wanted in docker.

**Comment by shymega** [comment]
Yeah. It sucks, but I think this issue should remain open until it's fixed, or a CLI flag is made available.

Part of me wonders, if it is fixable - built binaries in the shared $HOME on NixOS, will use the `/nix/store` `ld-linux`, and the built binaries in Docker/Podman will use their own `ld-linux`. So the problem will remain whatever happens - on at least one side, there'll be a `No such file or directory` executable error.

**Comment by 89luca89** [comment]
This has been fixed as of 5dd7b00e3f095e1976c0e2aa4f972efdcd6f4f3a

These integrations make little sense, as the target of
distrobox is to integrate user's data in a new userland, not
bring over host's userland inside

Maintainers will be able to reintroduce those mounts by using
a custom configuration if needed

**Comment by shymega** [comment]
@89luca89 Thanks, it's working for me now. Only in Podman, though, not Docker.

This should mean I can do my PlatformIO and ESP32 development now!

Thanks again! Stellar job.
